x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/AggregationResultUtils.java

```diff
@@ -334,6 +341,32 @@ public Object value(Aggregation agg, Map<String, String> fieldTypeMap, String lo
         }
     }
 
+    static class RangeAggExtractor implements AggValueExtractor {
+        @Override
+        public Object value(Aggregation agg, Map<String, String> fieldTypeMap, String lookupFieldPrefix) {
```

Comment:
I am not sure this is extracting the sub-aggs. 

For multi-bucket, we should extract sub aggs as well.

Here is a `terms` with sub-aggs:

```
POST _transform/_preview
{
  "source": {
    "index": "kibana_sample_data_flights"
  },
  "pivot": {
    "group_by": {
      "dest": {"terms": {
        "field": "DestAirportID"
      }}
    }, 
    "aggregations": {
      "t": {
        "terms": {
          "field": "OriginCountry",
          "size": 1
        },
        "aggs": {
          "avg": {
            "avg": {
              "field": "DistanceMiles"
            }
          }
        }
      }
    }
  }
}
```
what a bucket looks like:
```
{
      "t" : {
        "AE" : {
          "avg" : 8160.76904296875
        }
      },
      "dest" : "ABQ"
    }
```

But when I try with range:

```
{
  "source": {
    "index": "kibana_sample_data_flights"
  },
  "pivot": {
    "group_by": {
      "dest": {
        "terms": {
          "field": "DestAirportID"
        }
      }
    },
    "aggregations": {
      "r": {
        "range": {
          "field": "AvgTicketPrice",
          "keyed": true,
          "ranges": [
            {
              "to": 500
            },
            {
              "from": 500,
              "to": 700
            },
            {
              "from": 700
            }
          ]
        },
        "aggs": {
          "avg": {
            "avg": {
              "field": "DistanceMiles"
            }
          }
        }
      }
    }
  }
}
```
I still get just counts:
```
{
      "r" : {
        "500-700" : 0,
        "*-500" : 0,
        "700-*" : 2
      },
      "dest" : "ABQ"
    }
```

Reply:
Good spot!
I've just fixed it by reusing code for `MultiBucketsAggExtractor`.